nasa/earthdata-search
# EDSC-3797: Added an option for concatenation for custom downloads via Harmony

# Overview
### What is the feature?
Added an option for concatenation for custom downloads via Harmony
Please summarize the feature or fix.
Added checkbox for concatenation option as well as setting the checkbox to be checked if defaultConcatenation is true in the cmr request.

### What is the Solution?

- Added a new accessMethod options associated with the Harmony service.
  - supportsConcatenation (whether the option should show up)
  - defaultConcatenation (whether the option should be checked by default)
- Implemented the new checkbox in AccessMethod
- Implemented changes in the constructOrderPayload so concatenate is set to true if enableConcatenateDownload is true.

### What areas of the application does this impact?

List impacted areas.

# Testing

### Reproduction steps

- **Environment for testing: UAT**
- **Collection to test with: C1245618475-EEDTEST** 

1. Navigate to (http://localhost:8080/search?ee=uat)
2. Put in C1245618475-EEDTEST as the collection in the search box and search.
3. Click on the collection and then select 6 granules before clicking download.
4. Click on the 2nd radio button in the data access methods list
5. Observe the combine data customization option come up with default = false so the combine data option should be off.
6. click on the checkbox and then click download.
7. Observe the download going through successfully.

### Attachments
<img width="1609" alt="Screenshot 2023-11-09 at 1 22 25 PM" src="https://github.com/nasa/earthdata-search/assets/8591291/d6bbdd18-292c-4606-b2ae-50857c8b0d53">

# Checklist
